network/dag/interface.go

```diff
@@ -32,38 +33,47 @@ const AnyPayloadType = "*"
 var errRootAlreadyExists = errors.New("root transaction already exists")
 var errNoClockValue = errors.New("missing clock value")
 
-// DAG is a directed acyclic graph consisting of nodes (transactions) referring to preceding nodes.
-type DAG interface {
-	// Observable allows observers to be notified when a transaction is added to the DAG.
-	Observable
-	// Add adds one or more transactions to the DAG. If it can't be added an error is returned. Nil entries are ignored.
-	Add(ctx context.Context, transactions ...Transaction) error
-	// Walk visits every node of the DAG, starting at the given hash working its way down each level until every leaf is visited.
-	// when startAt is an empty hash, the walker starts at the root node.
-	// The walker will resolve the given starting hash to a clock value.
-	// The walk will be clock based so some transactions may be revisited due to existing branches.
-	// Precautions must be taken to handle revisited transactions.
-	Walk(ctx context.Context, visitor Visitor, startAt hash.SHA256Hash) error
+// State represents the Node transactional state. Mutations are done via this abstraction layer.
+// Notifications are also done via this layer
+type State interface {
+	// PayloadStore interface for wrapping payloadStore behind TX
+	PayloadStore
+	core.Diagnosable
+
+	// Add a transactions to the DAG. If it can't be added an error is returned.
+	// The payload may be passed as well. Allowing for better notification of observers
+	Add(ctx context.Context, transactions Transaction, payload []byte) error
 	// FindBetween finds all transactions which signing time lies between startInclude and endExclusive.
 	// It returns the transactions in DAG walking order.
 	FindBetween(ctx context.Context, startInclusive time.Time, endExclusive time.Time) ([]Transaction, error)
-	// Get retrieves a specific transaction from the DAG. If it isn't found, nil is returned.
-	Get(ctx context.Context, ref hash.SHA256Hash) (Transaction, error)
 	// GetByPayloadHash retrieves all transactions that refer to the specified payload.
 	GetByPayloadHash(ctx context.Context, payloadHash hash.SHA256Hash) ([]Transaction, error)
+	// GetTransaction returns the transaction from local storage
+	GetTransaction(ctx context.Context, hash hash.SHA256Hash) (Transaction, error)
```

Comment:
stick these functions in a DAG interface (like we had), since the interface applies to both `bboltDAG` and `state` (since you're delegating the calls).

Reply:
the bboltDAG impl is now only a concrete implementation. It no longer implements an interface. There's also no longer a need to mock it since all other components call the State interface.